crates/ruff_benchmark/benches/red_knot.rs

```diff
@@ -160,17 +236,23 @@ fn benchmark_cold(criterion: &mut Criterion) {
 
 #[track_caller]
 fn assert_diagnostics(db: &dyn Db, diagnostics: Vec<Box<dyn Diagnostic>>) {
-    let normalized: Vec<_> = diagnostics
-        .into_iter()
-        .map(|diagnostic| {
-            diagnostic
-                .display(db.upcast())
-                .to_string()
-                .replace('\\', "/")
-        })
-        .collect();
+    // We disable colors here so that our diagnostics don't include
+    // ANSI escape codes.
+    //
+    // NOTE: I kinda feel like this is a sub-optimal way of going about
+    // this. It just feels bad to be reaching in and doing global
+    // state. I feel like it would be better to have a more explicit
+    // configuration step before emitting diagnostics, but I think that
+    // requires more design.
+    colored::control::set_override(false);
 
-    assert_eq!(&normalized, EXPECTED_DIAGNOSTICS);
+    for (diagnostic, &expected) in diagnostics.into_iter().zip(EXPECTED_DIAGNOSTICS) {
+        let got = diagnostic
+            .display(db.upcast())
+            .to_string()
+            .replace('\\', "/");
+        assert_eq!(got, expected);
```

Comment:
Given that we have the diagnostic data structures here, why should we render them using `display` at all, now that that representation is so inconvienent for comparison and update? (Not that it was ever convenient.) Couldn't we just compare against, like, a tuple of the key fields instead?

Reply:
Yeah I'll look into this. I was mostly just looking to do the most minimal thing possible, but yeah, if you're cool with switching this out to just doing a comparison of key fields (range and diagnostic id I guess) then that seems fine.

Reply:
Yeah that seems definitely better to me -- the full diagnostic strings look like they'd be an absolute nightmare to update without help from `insta`.